packages/client/hmi-client/src/services/model-configurations.ts

```diff
@@ -235,3 +237,47 @@ export function getMissingInputAmount(modelConfiguration: ModelConfiguration) {
 	});
 	return missingInputs;
 }
+
+export function getMmtParameters(modelConfiguration, mmt, mmtParams) {
+	const collapsedParameters = collapseParameters(mmt, mmtParams);
+	const parameters = getParameters(modelConfiguration);
+	return Array.from(collapsedParameters.keys()).map((id) => {
+		const childIds = collapsedParameters.get(id) ?? [];
+		const childParameters = childIds
+			.map((childId) => parameters.find((p) => p.referenceId === childId))
+			.filter(Boolean) as ParameterSemantic[];
+
+		const parameter = {
+			default: false,
+			distribution: {
+				type: DistributionType.Constant,
+				parameters: { value: 0 }
+			},
+			referenceId: id,
+			type: SemanticType.Parameter,
+			source: mmt.annotations.name
+		};
+
+		return childParameters.length ? { ...childParameters[0] } : parameter;
```

Comment:
The function name is a bit confusing ... you are not "getting MMT parameters" per se, but looks like you are trying to merge two parameter lists? If so then following my other comment - don't use collapseXYZ functions, it will give you the wrong number of parameters, and also the wrong results. The question is then what are the two lists you are looking for?

Also, for operations dealing with two lists, it may be preferable to create a map instead of using "find()". The map will be a lot more performant.

For example:

```
const master = [ {k:1, v:1}, {k:2, v:2}, {k:3, v:3}, {k:4, v:4}]
const override = [ {k:1, v:9}, {k:3, v:99}, {k:9, v:3}]

const map = new Map();
override.forEach(d => {
	map.set(d.k, d);
});

const mergedList = master.map(d => {
	if (map.has(d.k)) {
		return map.get(d.k);
	}
	return d;
});

>> [ { k: 1, v: 9 }, { k: 2, v: 2 }, { k: 3, v: 99 }, { k: 4, v: 4 } ]
```

Reply:
switch the `find()` to a `map()`